Wrap all StdHook methods

Wauplin opened the pull request:
Hi !

First of all, thanks a lot for this package and for the Deepkit software in general ! We are using it to share experiments internally in our team and it helps a lot :smiley: 

I encountered an issue when trying to debug my code using [pdb](https://docs.python.org/3/library/pdb.html). It has to do with the terminal (stdout) that is catched around the `StdHook` wrapper that logs messages to Deepkit. When I use pdb, I cannot use the arrows to retrieve previous expressions. It is the same issue as described [in this issue](https://github.com/allenai/allennlp/issues/2176) (even though it's a different package and the source of the problem is different).

I have made a workaround to redirect all calls to `StdHook` object to the inner object `self.s`, except for the `write` method which is the one that needs to be overloaded. Maybe this slight improvement can be useful for other people :slightly_smiling_face:

Commit:
Wrap all StdHook methods

marcj commented:
Thanks @Wauplin! :) Which python version do you use? I added comments to the changes itself, as I'm not sure about the fix itself yet.

marcj commented on deepkit/__init__.py:
```diff
@@ -59,19 +59,19 @@ class StdHook:
 
+        def __getattr__(self, name):
+            if name == "s":
+                super(StdHook, self).__getattr__(name)
```
shouldn't this have a `return` statement? Or what is this supposed to do? `super(StdHook)` doesn't look right to me since StdHook has no parent class? I'm not sure I understand this fix.

```python
return super(StdHook, self).__getattr__(name)
```

Commit:
Add comments

Wauplin commented:
Hi @marcj, thanks for the quick feedback

> Which python version do you use?
I'm using Python 3.8.10 and I have to admit, I didn't try any other version.

> shouldn't this have a return statement?
Yes, my mistake.

> Or what is this supposed to do? super(StdHook) doesn't look right to me since StdHook has no parent class?
I have changed a bit the fix and added some comments with some references to explain basically what's happening. In fact, `super(StdHook)` is just returning the base class `object` which implements the base `__setattr__`.
